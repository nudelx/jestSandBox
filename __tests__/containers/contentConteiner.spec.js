import React from  'react'
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme'
import ContentContainer from '../../src/containers/ContentContainer'
import PageContent from '../../src/componets/pageContent'
import toJson from 'enzyme-to-json';
import enzymeSerializer from 'enzyme-to-json/serializer';
import configureStore from 'redux-mock-store'

const initialState = { content: 'test_1', movies: {movie:'test'}}
const mockStore = configureStore()
let store = mockStore(initialState)
let container = null

describe('ContentContainer test ', () => {

  beforeEach(()=>{
    store = mockStore(initialState)
    container = shallow(<ContentContainer store={store} /> )
 })

  it('should render without crashing', () => {
    expect(toJson(container)).toMatchSnapshot()
  })

  it('should get props from store  ', () => {
    expect(container.prop('content')).toEqual(initialState.content)
    expect(container.prop('movies')).toEqual(initialState.movies)
  })

})

describe('ContentContainer mount ', () => {
  beforeEach(()=>{
    store = mockStore(initialState)
    container = mount(<ContentContainer store={store}/>)
 })

  it('should render PageContent  ', () => {
    const component = container.find(ContentContainer)
    expect(toJson(component)).toMatchSnapshot()
  })
})
