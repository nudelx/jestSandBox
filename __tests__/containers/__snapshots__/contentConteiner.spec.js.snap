// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentContainer mount  should render PageContent   1`] = `
<Connect(ContentContainer)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ContentContainer
    content="test_1"
    dispatch={[Function]}
    movies={
      Object {
        "movie": "test",
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <PageContent
      content="test_1"
      movies={
        Object {
          "movie": "test",
        }
      }
    >
      <div
        className="page-content"
      >
        <EmptyCard>
          <div>
            <div
              className="movie-title"
            />
            <div
              className="movie-data-empty"
            />
          </div>
        </EmptyCard>
      </div>
    </PageContent>
  </ContentContainer>
</Connect(ContentContainer)>
`;

exports[`ContentContainer test  should render without crashing 1`] = `
<ContentContainer
  content="test_1"
  dispatch={[Function]}
  movies={
    Object {
      "movie": "test",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
